src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/MalformedRoundaboutCheck.java

```diff
@@ -1,36 +1,47 @@
 package org.openstreetmap.atlas.checks.validation.linear.edges;
 
-import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.HashSet;
-import java.util.LinkedList;
 import java.util.List;
 import java.util.Optional;
-import java.util.Queue;
 import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Stream;
 
 import org.openstreetmap.atlas.checks.base.BaseCheck;
 import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.geography.PolyLine;
+import org.openstreetmap.atlas.geography.Polygon;
 import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
 import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.items.Node;
+import org.openstreetmap.atlas.geography.atlas.items.Route;
+import org.openstreetmap.atlas.geography.atlas.walker.SimpleEdgeWalker;
+import org.openstreetmap.atlas.tags.BridgeTag;
+import org.openstreetmap.atlas.tags.HighwayTag;
 import org.openstreetmap.atlas.tags.ISOCountryTag;
 import org.openstreetmap.atlas.tags.JunctionTag;
+import org.openstreetmap.atlas.tags.LayerTag;
+import org.openstreetmap.atlas.tags.TunnelTag;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
 import org.openstreetmap.atlas.utilities.configuration.Configuration;
 
 /**
- * This check flags roundabouts where the directionality is opposite to what it should be, where the
- * roundabout is multi-directional, or where the roundabout has incorrect geometry (concave).
+ * This check flags roundabouts where the directionality is opposite to what it should be, the
+ * roundabout is multi-directional, the roundabout is incomplete, part of the roundabout is not car
+ * navigable, or the roundabout has car navigable edges inside it.
  *
  * @author savannahostrowski
+ * @author bbreithaupt
  */
 
 public class MalformedRoundaboutCheck extends BaseCheck
 {
     private static final long serialVersionUID = -3018101860747289836L;
-    private static final String WRONG_WAY_INSTRUCTIONS = "This roundabout, {0,number,#}, is going the"
+    private static final String WRONG_WAY_INSTRUCTIONS = "This roundabout is going the"
             + " wrong direction, or has been improperly tagged as a roundabout.";
-    private static final String MULTIDIRECTIONAL_INSTRUCTIONS = "This roundabout, {0,number,#}, is"
-            + " multi-directional, or the roundabout has improper angle geometry.";
+    private static final String INCOMPLETE_ROUTE_INSTRUCTIONS = "This roundabout does not form a single, one-way, complete, car navigable route.";
+    private static final String ENCLOSED_ROADS_INSTRUCTIONS = "This roundabout has car navigable ways inside it.";
```

Comment:
Can you clarify why it's a problem that there are car navigable ways inside the roundabout? Does it make the roundabout improperly formed?

Reply:
If there are car navigable ways inside a roundabout it generally indicates the object is either a ring road or other larger set of roads, or a throughabout. None of these items should be tagged as roundabouts.